ployst/github/hooker.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+import hashlib
+import hmac
+import json
+
+
+DEBUG = True
+
+
+class GithubHookException(Exception):
+    pass
+
+
+def compute_github_signature(body, secret):
+    return hmac.new(secret, body, hashlib.sha1).hexdigest()
+
+
+class GithubHookHandler(object):
+    """
+    A general purpose-class to help build github hook handlers.
+
+    See https://developer.github.com/webhooks/ for Github's webhook spec.
+
+    NOTE: This module may be spin-off into a separate library.
+    """
+
+    @staticmethod
+    def get_org_repo(request):
+        try:
+            payload = json.loads(request.body)
+            url = payload['repository']['url']
+            org, repo = url.split('/')[-2:]
+            return org, repo
+        except ValueError:
+            raise GithubHookException("Invalid payload")
+
+    def __init__(self, secret, request):
+        hub_sig = request.META['HTTP_X_HUB_SIGNATURE']
+        payload = request.body
+
+        if not self.validate_hook_post(payload, secret, hub_sig):
+            raise GithubHookException("Incorrect signature")
+
+        self.payload = json.loads(request.body)
+        self.event = request.META['HTTP_X_GITHUB_EVENT']
+
+        if DEBUG:
+            filename = 'github-hook-{}.json'.format(datetime.now().isoformat())
+            with open(filename, 'w') as f:
+                f.write(request.body)
```

Comment:
This is something that I'm adding temporarily as a way to collect payload samples from all github activity.
It may be handy to keep something like this in the future?